src/benchmark_ocl.cpp

```diff
@@ -144,125 +119,106 @@ int main(void)
 		auto dt_tmp = dt;
-		err = clSetKernelArg(kernel, 6, sizeof(real_t), static_cast<const void*>(&dt_tmp));
-		ocl_error_handler(err, "clSetKernelArg(6)");
+		err = kernel.setArg(6, dt_tmp);
+		noma::ocl::error_handler(err, "kernel.setArg(6)");
 
 		return kernel;
 	}; // prepare_kernel
 
 	auto write_hamiltonian = [&]()
 	{
-		err = clEnqueueWriteBuffer(CLU_DEFAULT_Q, hamiltonian_ocl, CL_TRUE, 0, size_hamiltonian_byte, hamiltonian, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueWriteBuffer(hamiltonian)");
+		err = ocl_helper.queue().enqueueWriteBuffer(hamiltonian_ocl, CL_TRUE, 0, size_hamiltonian_byte, hamiltonian);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(hamiltonian)");
 	}; // write_hamiltonian
 
 	// lambda to write data to the device
 	auto write_sigma = [&]()
 	{
 		// write data to device
-		err = clEnqueueWriteBuffer(CLU_DEFAULT_Q, sigma_in_ocl, CL_TRUE, 0, size_sigma_byte, sigma_in, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueWriteBuffer(sigma_in_ocl)");
-		err = clEnqueueWriteBuffer(CLU_DEFAULT_Q, sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueWriteBuffer(sigma_out_ocl)");
+		err = ocl_helper.queue().enqueueWriteBuffer(sigma_in_ocl, CL_TRUE, 0, size_sigma_byte, sigma_in);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(sigma_in_ocl)");
+		err = ocl_helper.queue().enqueueWriteBuffer(sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(sigma_out_ocl)");
 	}; // write_sigma
 
 	// lambda to get the result from the device and compare it with the reference
 	auto read_and_compare_sigma = [&]()
 	{
 		// read data from device	
-		err = clEnqueueReadBuffer(CLU_DEFAULT_Q, sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueReadBuffer(sigma_out_ocl)");
+		err = ocl_helper.queue().enqueueReadBuffer(sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out);
+		noma::ocl::error_handler(err, "clEnqueueReadBuffer(sigma_out_ocl)");
 		// compute deviation from reference	(small deviations are expected)
 		deviation = compare_matrices(sigma_out, sigma_reference_transformed, dim, num);
 		std::cerr << "Deviation:\t" << deviation << std::endl;
 	}; // read_and_compare_sigma
 
 	// Lambda to: transform memory, benchmark, compare results
 	// NOTE:
-	// typedef struct
+	// struct noma::ocl::nd_range
 	// {
-	//     cl_uint   dim; 
-	//     size_t    global[3];
-	//     size_t    local[3];
-	//     size_t    offset[3];
-	// } clu_nd_range;
+	//     cl::NDRange offset;
+	//     cl::NDRange global;
+	//     cl::NDRange local;
+	// };
 	auto benchmark = [&](const std::string& file_name, const std::string& kernel_name,
-	                     const std::string& compile_options, size_t vec_length, clu_nd_range range,
+	                     const std::string& compile_options, size_t vec_length, noma::ocl::nd_range& range,
```

Comment:
nd_range can be const here

Reply:
Done with a758cba0fa201d64c477ff298ffa7ea784117a09.